packages/core/src/common/utils.ts

```diff
@@ -104,3 +104,36 @@ export function throttleEvent(target: EventTarget, eventName: string, newEventNa
     target.addEventListener(eventName, func);
     return func;
 };
+
+/**
+ * Throttle a callback by wrapping it in a `requestAnimationFrame` call. Returns the throttled
+ * function.
+ * @see https://www.html5rocks.com/en/tutorials/speed/animations/
+ */
+export function throttleReactEventCallback(
+    callback: (event: React.SyntheticEvent<any>, ...otherArgs: any[]) => any,
+    preventDefault?: boolean,
```

Comment:
Another option here is `{ preventDefault?: boolean }?: IThrottledReactEventOptions`. That lets us explicitly write `{ preventDefault: true }` when invoking this function (right now, I'm adding an inline comment like `/* preventDefault */ true`.